presto-main/src/main/java/com/facebook/presto/server/AsyncPageTransportServlet.java

```diff
@@ -104,10 +105,10 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response)
 
         AsyncContext asyncContext = request.startAsync(request, response);
 
+        // wait time to get results
         Duration waitTime = randomizeWaitTime(DEFAULT_MAX_WAIT_TIME);
-        asyncContext.setTimeout(waitTime.toMillis() + PAGE_RESPONSE_WAIT_TIME.toMillis());
+        asyncContext.setTimeout(waitTime.toMillis() + pageTransportTimeout.toMillis());
 
-        // set up async listener
```

Comment:
accidentally removed this line ? :)

Reply:
`asyncContext.addListener()` is clear enough :)